agent/src/bpf/audit.bpf.c

```diff
@@ -49,6 +55,8 @@ record_new_context (struct pt_regs *ctx)
 			 sizeof(*event),
 			 context);
   event->parent = parent;
+  if (build_id.status & BPF_STACK_BUILD_ID_VALID)
+    __builtin_memcpy (event->origin, build_id.build_id, sizeof(event->origin));
```

Comment:
will that overflow for `--build-id=sha256`?

Reply:
It seems the kernel side also only supports 20 bytes: https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/bpf.h#L1308, so maybe this isn't an issue?

Would be interesting to test and see what happens with longer build IDs, though.